solidity-checks/src/main/java/org/sonarsource/solidity/checks/ContractNotEmptyCheck.java

```diff
@@ -0,0 +1,18 @@
+package org.sonarsource.solidity.checks;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.check.Rule;
+import org.sonarsource.solidity.frontend.SolidityParser.ContractDefinitionContext;
+
+@Rule(key = ContractNotEmptyCheck.RULE_KEY)
+public class ContractNotEmptyCheck extends IssuableVisitor {
+
+  public static final String RULE_KEY = "ExternalRule2";
+
+  @Override
+  public ParseTree visitContractDefinition(ContractDefinitionContext ctx) {
+    if (ctx.contractPart().isEmpty())
+      ruleContext().addIssue(ctx.getStart(), ctx.identifier().getStart(), "Contract should not be empty", "ExternalRule2");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use already-defined constant 'RULE_KEY' instead of duplicating its value here. ([squid:S1192](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1192))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=25&issues=AWRguWtHMt9UOhrw62Qm&open=AWRguWtHMt9UOhrw62Qm)

[ISSUE_KEY]: AWRguWtHMt9UOhrw62Qm